webrtc-c/canary/src/Config.cpp

```diff
@@ -201,4 +221,108 @@ STATUS Config::init(INT32 argc, PCHAR argv[])
     return retStatus;
 }
 
+VOID Config::print()
+{
+    DLOGD("\n\n"
+          "\tChannel Name  : %s\n"
+          "\tRegion        : %s\n"
+          "\tClient ID     : %s\n"
+          "\tRole          : %s\n"
+          "\tTrickle ICE   : %s\n"
+          "\tUse TURN      : %s\n"
+          "\tLog Level     : %u\n"
+          "\tLog Group     : %s\n"
+          "\tLog Stream    : %s\n"
+          "\tDuration      : %lu seconds\n"
+          "\tIteration     : %lu seconds\n"
+          "\n",
+          this->channelName.value, this->region.value, this->clientId.value, this->isMaster.value ? "Master" : "Viewer",
+          this->trickleIce.value ? "True" : "False", this->useTurn.value ? "True" : "False", this->logLevel.value, this->logGroupName.value,
+          this->logStreamName.value, this->duration.value / HUNDREDS_OF_NANOS_IN_A_SECOND,
+          this->iterationDuration.value / HUNDREDS_OF_NANOS_IN_A_SECOND);
+}
+
+VOID jsonString(PBYTE pRaw, jsmntok_t token, Config::Value<Config::String>* pResult)
+{
+    UINT32 tokenLength = (UINT32)(token.end - token.start);
+
+    SNPRINTF(pResult->value, MIN(tokenLength + 1, ARRAY_SIZE(pResult->value)), "%s", (const CHAR*) pRaw + token.start);
+    pResult->initialized = TRUE;
+}
+
+VOID jsonBool(PBYTE pRaw, jsmntok_t token, Config::Value<BOOL>* pResult)
+{
+    Config::Value<Config::String> raw;
+
+    jsonString(pRaw, token, &raw);
+    pResult->value = strtobool((const CHAR*) &raw.value);
+    pResult->initialized = TRUE;
+}
+
+VOID jsonUint64(PBYTE pRaw, jsmntok_t token, Config::Value<UINT64>* pResult)
+{
+    Config::Value<Config::String> raw;
+
+    jsonString(pRaw, token, &raw);
+    STRTOUI64((PCHAR) raw.value, NULL, 10, &pResult->value);
+    pResult->initialized = TRUE;
+}
+
+STATUS Config::initWithJSON(PCHAR filePath)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 size;
+    jsmn_parser parser;
+    int r;
+    BYTE raw[MAX_CONFIG_JSON_FILE_SIZE];
+    Config::Value<UINT64> logLevel64;
+
+    CHK_STATUS(readFile(filePath, TRUE, NULL, &size));
+    CHK_ERR(size < 1024, STATUS_INVALID_ARG_LEN, "File size too big. Max allowed is 1024 bytes");
```

Comment:
MAX_CONFIG_JSON_FILE_SIZE instead of 1024. Same with the printing

Reply:
Ouch, my bad.

Reply:
Updated.